animoca/revv-ethereum-contracts
# Amoy deployment

Work in progress PR

- add missing artifacts from previous versions
  - add ethereum-universal-forwarder v0.1.4 as a dependency
- add deploy scripts for the different contracts
- add configs for deployment on Sepolia and Amoy
- add tasks to generate and set the merkle tree on REVVMotorsportShardClaim
- add deployment and REVVMotorsportShardClaim config instructions to README

## Bridge REVV token from Sepolia to Amoy
- Go to https://portal.polygon.technology/bridge
- Switch to testnet (link in the bottom left corner)
- Select transfer from "Sepolia"
- Change the token to transfer, it should open a modal
- Input "0x8bfd3020b3aeed17a96be47b181950fa1f02dc31" in the search
- If the REVV token was not already imported, Click "Add REVV token" at the bottom of the modal. Otherwise select "REVV"
- Select transfer to "Amoy"
- Enter amount to transfer
- Confirm 🎊 

## Sepolia deployment

- ForwarderRegistry: [0x71beE13E92e64701b78f059801A70857E4683ACd](https://sepolia.etherscan.io/address/0x71beE13E92e64701b78f059801A70857E4683ACd#code)
- REVV: [0x8bfd3020b3aeed17a96be47b181950fa1f02dc31](https://sepolia.etherscan.io/address/0x8bfd3020b3aeed17a96be47b181950fa1f02dc31#code)
- REVVMotorsportShard: [0x493E3b95BBdB61a07D16fA78E17220d9e25dfEd5](https://sepolia.etherscan.io/address/0x493E3b95BBdB61a07D16fA78E17220d9e25dfEd5#code)

## Amoy deployment

- ForwarderRegistry: [0x71beE13E92e64701b78f059801A70857E4683ACd](https://amoy.polygonscan.com/address/0x71beE13E92e64701b78f059801A70857E4683ACd#code)
- UniversalForwarder: [0x56133a31F4cD01EDe8c6EE780071c021bd324aD2](https://amoy.polygonscan.com/address/0x56133a31F4cD01EDe8c6EE780071c021bd324aD2#code)
- PolygonREVV: [0x7B7aE00a388D3147Ccdf27C37317a80A00b080E7](https://amoy.polygonscan.com/address/0x7B7aE00a388D3147Ccdf27C37317a80A00b080E7#code)
- SessionsManager Free: [0x307D2AE9472f311d0a413F048DA1B4Ed2F19333c](https://amoy.polygonscan.com/address/0x307D2AE9472f311d0a413F048DA1B4Ed2F19333c#code)
- SessionsManager 1 REVV: [0xd7ee0eADb283eFB6d6e628De5E31A284183f4EDf](https://amoy.polygonscan.com/address/0xd7ee0eADb283eFB6d6e628De5E31A284183f4EDf#code)
- SessionsManager 2 REVV: [0xe6f23ecc9f317074c3f87c8674389fd8ed2c08ca](https://amoy.polygonscan.com/address/0xe6f23ecc9f317074c3f87c8674389fd8ed2c08ca#code)
- SessionsManager 3 REVV: [0xC251bC036c05e98cd7887B3dc2b63d4cc2A252fa](https://amoy.polygonscan.com/address/0xC251bC036c05e98cd7887B3dc2b63d4cc2A252fa#code)
- SessionsManager 4 REVV: [0x99079f610a8CD19C7A05a761B0d4a73A67ee1eD3](https://amoy.polygonscan.com/address/0x99079f610a8CD19C7A05a761B0d4a73A67ee1eD3#code)
- SessionsManager 5 REVV: [0x79780B6d60BE56467FEfd13c38B1A22D37CCED7A](https://amoy.polygonscan.com/address/0x79780B6d60BE56467FEfd13c38B1A22D37CCED7A#code)
- REVVRacingNFT: [0x6DE4F7480aDFEc50a487DC838120f5dd9eb08414](https://amoy.polygonscan.com/address/0x6DE4F7480aDFEc50a487DC838120f5dd9eb08414#code)
- PolygonREVVMotorsportShard: [0xe249c50DfBd4c3eCcf7E00C9C068B8Aa32BB9a22](https://amoy.polygonscan.com/address/0xe249c50DfBd4c3eCcf7E00C9C068B8Aa32BB9a22#code)
- REVVMotorsportShardClaim: [0x90bAD614b635c2537bFe679232B672e8b07281d4](https://amoy.polygonscan.com/address/0x90bAD614b635c2537bFe679232B672e8b07281d4#code)

## TODO
- Mint `REVVRacingNFT` tokens
  - Requires the sender to have the `MINTER_ROLE`. Who needs this role?
